src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,108 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import java.io.IOException;
+
+public class Storage {
+
+    private String pathName;
+    private String fileName;
+
+    public Storage(String pathName, String fileName) {
+        this.pathName = pathName;
+        this.fileName = fileName;
+    }
+
+    /**
+     * Returns a TaskList that stores tasks retrieved from the data/duke.txt if the file exits
+     *
+     * @return a TaskList with tasks stored in the txt file.
+     */
+    public TaskList load() {
+        TaskList tasks = new TaskList( new ArrayList<Task>() );
+        try {
+            File file = new File("data/duke.txt");
+            Scanner sc = new Scanner(file);
+            while (sc.hasNext()) {
+                String previousTask = sc.nextLine();
+                String[] words = previousTask.split(" @ ", 0);
+                if (words[0].equals("D")) {
+                    Deadline previousDeadline = new Deadline(words[2], words[3]);
+                    if (words[1].equals("1")) {
+                        previousDeadline.setIsDone();
+                    }
+                    tasks.add(previousDeadline);
+                } else if (words[0].equals("T")) {
+                    ToDo previousToDo = new ToDo(words[2]);
+                    if (words[1].equals("1")) {
+                        previousToDo.setIsDone();
+                    }
+                    tasks.add(previousToDo);
+                } else if (words[0].equals("E")) {
+                    Event previousEvent = new Event(words[2], words[3]);
+                    if (words[1].equals("1")) {
+                        previousEvent.setIsDone();
+                    }
+                    tasks.add(previousEvent);
+                }
+            }
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+        return tasks;
+    }
```

Comment:
Perhaps can abstract reading task from line to avoid duplicate check of task status and set the task as done?

Reply:
sure thanks